.github/workflows/publish-images.yaml

```diff
@@ -0,0 +1,64 @@
+---
+name: Publish Images
+
+on:
+  workflow_call:
+
+jobs:
+  push-image:
+    strategy:
+      matrix:
+        image: ['controller', 'executor']
+    runs-on: ubuntu-latest
+    permissions:
+      actions: none
+      checks: none
+      contents: none
+      deployments: none
+      issues: none
+      packages: write
+      pull-requests: none
+      repository-projects: none
+      security-events: none
+      statuses: none
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v3.0.2
+
+      - name: Login to GitHub
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - id: docker_meta
```

Comment:
Any chance i can have a quick explanation how this worls - and how it's works out latest vs tags ...